# Tidak Kenal maka Tidak Sayang

- Nama : Ferly Jeremi Purnawan Apiang
- NRP : 1973005


# Training Classifier Random Forest

Pemanggilan Dataset Breast Cancer dengan menggunakan Training Classifier **Random Forest**

## Load Breast Cancer Dataset

In [8]:
from sklearn.datasets import load_breast_cancer

X, y = load_breast_cancer(return_X_y=True)

print(f'Dimensi Feature: {X.shape}')
print(f'Class: {set(y)}')

Dimensi Feature: (569, 30)
Class: {0, 1}


- Pertama Lakukan pemanggilan dengan memanggil datasetnya (import load_breast_cancer)
- Panggil loadnya dengan menanampungnya ke dalam variable yang saya beri nama X dan y
- Lalu saya ingin mengetahui dimensi dari datasetnya saya panggil Print X.shape dan set{y} untuk class.

Hasil yang dikeluarkan yaitu Data Feature adalah (569, 30) terdapat 569 Baris dan 30 kolom. Lalu dengan class pada target labelnya terdapat {0, 1}

## Training dan Testing Dataset

Kita akan mencoba melakukan ke dalam sebuah training set dan testing set

In [9]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X,
                                                    y,
                                                    test_size=0.25,
                                                    random_state=0)

- Penggunaan Model Sklearn "from sklearn.model_selection lalu untuk Testing kita import train_test_split".
- Lalu kita panggil train_test_split nya, lalu kita sertakan X dan y (variable masing-masing) sebagai parameternya.
- Lalu terkait parameter test_size nya kita set sebagai 0.25 yang artinya disini testing set nya akan menempati 25% dari total keseluruhan dataset yang kita miliki.
- Lalu berikutnya untuk random_state kita beri nilai 0.
- Pemanggilan fungsi train_test_split ini akan menghasilkan 4 kumpulan data, keempatnya perlu ditampung ke dalam 4 buah variabel yaitu X_train, X_test, y_train, dan y_test.

## Classification **RandomForestClassifier** dengan Sklearn

menerapkan Random Forest untuk melakukan klasifkasi Breast Cancer.

In [10]:
from sklearn.ensemble import RandomForestClassifier

model = RandomForestClassifier(n_estimators=150, 
                               random_state=0)

model.fit(X_train, y_train)

RandomForestClassifier(n_estimators=150, random_state=0)

- Pertama-tama kita akan mengimport terlebihi dahulu modulnya dengan memanggil method *'from sklearn.ensemble import RandomForestClassifier'*.
- Selanjutnya kita akan bentuk objek dari *'RandomForestClassifier'* ini dengan menyertakan 2 buah parameter yaitu *'(n_estimators=150)'* yang untuk kasus kali ini kita melakukan training 150 buah model.
- Lalu agar eksperimen kita ini bisa kita replicate dengan hasil yang konsisten, kita juga perlu mengeset random statenya yang dalam kasus kita kali ini adalah *'(random_state=0)'*. 
- Selanjutnya objek yang terbentuk tersbut akan kita tampung kedalam variabel *model*. 
- Untuk selanjutnya, model ini akan kita training dengan memanggil method *fit* dengan menyertakan parameter *'(X_train, y_train)'*.

# Evaluasi Model

Disini kita akan lakukan evaluasi performa dari model **Random Forest** yang baru saja kita training tadi dengan memanfaatkan **Classification Report**.

In [11]:
from sklearn.metrics import classification_report

y_pred = model.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.95      0.98      0.96        53
           1       0.99      0.97      0.98        90

    accuracy                           0.97       143
   macro avg       0.97      0.97      0.97       143
weighted avg       0.97      0.97      0.97       143



- Pertama-tam kita akan import terlebih dahulu modulnya dengan memanggil *'from sklearn.metrics import classification_report'*.
- Lalu berikutnya kita lakukan prediksi terhadap x_test dengan memanggil *'model.predict(X_test)'* yang akan kita tampung kedalam variabel *y_pred*.
- Untuk selanjutnya kita akan memanggil *'classification_report'* dengan menyertakan parameter *'(y_test, y_pred)'*.

Hasil output merupakan **classification report** sebagai bentuk evaluasi terhadap model **Random Forest** yang baru saja kita training sebelumnya, ditampilkan nilai *precision*, *recall*, serta *f1-score* nya.

In [12]:
from sklearn.metrics import classification_report
import numpy as np
import pandas as pd

from sklearn.datasets import make_classification
from sklearn.linear_model import LogisticRegressionCV
from sklearn.model_selection import train_test_split, GridSearchCV

In [13]:
X, y = make_classification(n_samples=1000, n_features=30,
                           n_informative=12,
                           n_clusters_per_class=1, n_classes=10,
                           class_sep=2.0, random_state=42)

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.3, stratify=y)


clf = LogisticRegressionCV(max_iter=1000, random_state=42).fit(X_train, y_train)



df = pd.DataFrame(classification_report(clf.predict(X_test), 
                                        y_test, digits=2,
                                        output_dict=True)).T

df['support'] = df.support.apply(int)

df.style.background_gradient(cmap='viridis',
                             subset=pd.IndexSlice['0':'1', :'f1-score'])

C:\Users\ferly\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:762: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
C:\Users\ferly\anaconda3\lib\site-packages\sklearn\linear_model\_logistic.py:762: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_

- langka pemanggilan ada banyak sekali tapi kalian bisa lihat saya sudah memisahkan apa-apa saja yang perlu diimport.
- Masuk untuk membuat classification Dengan menentukan samples saya memberi batas 1000 lalu, featuresnya saya beri jarak 30.
- df.suport.apply fungsinya untuk menentukan berapa kemungkinan orang yang terkena.

Ya jadi demikian Training Classifier menggunakan **Random Forest dengan Breast Cancer Dataset**